### PR TITLE
 devilutionx: 1.3.0 -> 1.4.0 

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/devilutionx/default.nix [hunk 1]
```diff
@@ -16,57 +18,75 @@
 
 let
   # TODO: submit a PR upstream to allow system copies of these libraries where possible
-  asio = fetchurl {
+
+  # fork with patches, far behind upstream
+  asio = fetchurl rec {
```
Comment (dtzWill) [inline pkgs/games/devilutionx/default.nix:23]: No need for `rec`.

File: pkgs/games/devilutionx/default.nix [hunk 2]
```diff
@@ -16,57 +18,75 @@
...
+    sha256 = "sha256-QMDcIZQ94i4VPVanmSxiGkKgxWx82DP4uE+Q5I2nU+o=";
+  };
+
+  # fork with patches, far behind upstream
+  libzt = fetchFromGitHub rec {
+    owner = "diasurgical";
+    repo = "libzt";
+    fetchSubmodules = true;
+    rev = "a34ba7f1cc2e41b05badd25d1b01fdc5fd2f4e02";
+    sha256 = "sha256-tyIPt+7OTi5W+5X/ixQP1puOKdhrqRZtpwXlWFBFpc8=";
```
Comment (dtzWill) [inline pkgs/games/devilutionx/default.nix:46]: (same for these as well)

File: pkgs/development/libraries/SDL_audiolib/default.nix [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    SDL2
+  ];
+
+  cmakeFlags = [
```
Comment (dtzWill) [inline pkgs/development/libraries/SDL_audiolib/default.nix:29]: Why are these all disabled? Is it possibly there'll be another user of SDL_audiolib that expects these?

Anyway curious if nothing else, thanks :).

File: pkgs/build-support/fetchurl/default.nix [hunk 4]
```diff
@@ -160,5 +160,5 @@ stdenvNoCC.mkDerivation {
   '';
 
   inherit meta;
-  inherit passthru;
+  passthru = { inherit url; } // passthru;
```
Comment (dtzWill) [inline pkgs/build-support/fetchurl/default.nix:163]: This seems like something to be discussed and not nonchalantly introduced updating a package.
I don't have a problem with it, but haven't really thought about it a lot.  Fetching `url` back does seem useful, and I see it helps a bit for how the devilutionx expression is structured.

Alternatively, it'd be easy enough to just introduce a helper function in devilutionx that either does effectively this (put `url` in `passthru`), or returns both the resulting store path and the fetched url together: `fetch = url: hash: { src = fetchurl { inherit url hash; }; inherit url; }; asio = fetch "https://..." "sha256-..."; ...` or maybe something little tidier :).  And in the substitutions each accesses `asio.url` and `asio.src`.

Anyway if fetchurl has a `.url` what you have sounds great, but `fetchurl` is not usually casually modified like this.

File: pkgs/development/libraries/SDL_audiolib/default.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    "-DUSE_DEC_BASSMIDI=OFF"
+    "-DUSE_DEC_WILDMIDI=OFF"
+    "-DUSE_DEC_ADLMIDI=OFF"
+  ];
+
+  meta = with lib; {
+    description = "Audio decoding, resampling and mixing library for SDL";
+    homepage = "https://github.com/realnc/SDL_audiolib";
+    license = licenses.lgpl3Plus;
+    maintainers = with maintainers; [ SuperSandro2000 ];
```
Comment (dtzWill) [inline pkgs/development/libraries/SDL_audiolib/default.nix:51]: Thanks! :+1:

**Review by dtzWill — CHANGES_REQUESTED** [review]
Thanks! And thanks for introducing and maintaining the `SDL_audiolib`! :tada: 

Minor nit re:`rec`, and not sure this motivates changing `fetchurl` although offhand I don't know what problems doing that might cause (but it's so critical to so much of nixpkgs, idk maybe causes evaluation memory usage to spike because of references to source URL's, etc).  Perhaps should be pulled out and given some light independently?

If nothing else, please flag such changes much more clearly.  This was not expected as part of a minor package update :).

File: pkgs/development/libraries/SDL_audiolib/default.nix (hunk 3 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/SDL_audiolib/default.nix:29]: Most of them draw in other libraries and I was to lazy which of these we already have and which would need to be packaged. The game does not need these and has the options itself disabled.

File: pkgs/build-support/fetchurl/default.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/build-support/fetchurl/default.nix:163]: Most other fetchers like fetchFromGitHub do similar things and the performance difference is negligible. You can already pass it through like ``fetchurl rec { passthru.url = url; }`` which is IMO a bit useless without a good default. Definitely changes nothing about how fetchurl works. Extracting it into another PR just creates overhead.

**Pull request merged**